dist/PublicLab.Grapher.js

```diff
@@ -235,10 +253,108 @@ function () {
+          _this.csvParser = new _CsvParser2.CsvParser(_this.csvFile, _this.elementId, "local");
         };
       }
     }
+  }, {
+    key: "handleFileSelectstring",
+    value: function handleFileSelectstring(val) {
+      // this.csvFileString = val;
+      console.log("value", val);
+      console.log("i am at csv string handler");
+      var csv_string = val.split("\n");
+      var mat = [];
+
+      for (var i = 0; i < csv_string.length; i++) {
+        if (csv_string[i] == "" || csv_string[i] == " ") {
+          continue;
+        }
+
+        var dataHash = Papa.parse(csv_string[i], {
+          dynamicTyping: true,
+          comments: true
+        });
+        mat[i] = dataHash['data'][0];
+      }
+
+      this.csvFile = mat;
+
+      document.getElementById(this.uploadButtonId).onclick = function (e) {
+        console.log("i am uploading");
+        console.log(this);
+        this.csvParser = new _CsvParser.CsvParser(this.csvFile, this.elementId, "csvstring");
+      };
+    }
+  }, {
+    key: "handleFileSelectGoogleSheet",
+    value: function handleFileSelectGoogleSheet(val) {
+      var headers_sheet = [];
+      var matrixComplete = [];
+      $.getJSON(val, function (data) {
+        var hashSheet = data.feed.entry;
+
+        for (var key in hashSheet) {
+          var h = hashSheet[key];
+
+          for (var headKey in h) {
+            if (headKey.slice(0, 4) == "gsx$") {
+              headers_sheet.push(headKey);
+            }
+          }
+
+          break;
+        }
+
+        for (var i = 0; i < headers_sheet.length; i++) {
+          matrixComplete[i] = [];
+        }
+
+        for (var i = 0; i < headers_sheet.length; i++) {
+          var j = 0;
+
+          for (var key in hashSheet) {
+            matrixComplete[i].push(hashSheet[key][headers_sheet[i]]["$t"]);
+            j++;
+          }
+        }
+
+        for (var i = 0; i < headers_sheet.length; i++) {
+          headers_sheet[i] = headers_sheet[i].slice(4, headers_sheet[i].length);
+        } // var totaldata=[headers_sheet,matrixComplete];
+        // this.sheetData=totaldata;
+
+
+        console.log(headers_sheet, "hh");
+      });
+      this.csvFile = [headers_sheet, matrixComplete];
+
+      document.getElementById(this.uploadButtonId).onclick = function (e) {
+        console.log("i am uploading");
+        console.log(this);
+        this.csvParser = new _CsvParser.CsvParser(this.csvFile, this.elementId, "googleSheet");
+      };
+    }
+  }, {
+    key: "receive",
+    value: function receive(vall) {
+      console.log(vall);
+      console.log("hurray!!");
+    }
+  }, {
+    key: "handleFileSelectremote",
+    value: function handleFileSelectremote(val) {
+      var proxyurl = "https://cors-anywhere.herokuapp.com/";
+      var url = val;
+      fetch(proxyurl + url).then(function (response) {
+        return response.text();
+      }).then(function (contents) {
+        return console.log(contents);
+      })["catch"](function () {
```

Comment:
I'm not completely familiar with this syntax but i think it might be: 
```suggestion
      })["catch"](function (e) {
      console.log('error!', e);
```

Reply:
@jywarren I think this should be automatically incorporated once @IshaGupta18 passes the error object below ([as I suggested](https://github.com/publiclab/simple-data-grapher/issues/45#issuecomment-501793787)) since the dist is built from that?
https://github.com/publiclab/simple-data-grapher/blob/46ff3bd268bf1e21cba88286917dd08a8c8fe983/src/View.js#L127